### PR TITLE
fix(@desktop/settings-contacts): it is not possible to add new contacts after the first one is added

### LINKED ISSUE #4658: It is not possible to add new contacts after the first one is added
**Steps:**

1. Go to Settings / Add new contact
2. Write contact key

**Expected:**
Contact info is displayed and the add icon is shown

**Actual:**
Contact info is displayed but add icon is not shown. Instead the already added icon is shown. 

_NOTE:_ It happens after a first contact is added since the first contact can be added correctly.

https://images.zenhubusercontent.com/261270348/1de9170a-090e-4fe2-bb32-c04fd2566b07/screen_recording_2022_02_02_at_09_27_08.mov

### PR DESCRIPTION
**Pull request by saledjenic**
@Khushboo-dev-cpp please check in the app do changes I did here (in `contacts/async_tasks.nim`) affect somehow what you fixed in your PR regarding searching for user's address when you want to send him a transaction. If it does affect please give me the steps how to reproduce it, since I tried and it looked ok for me.

Fixes #4658

### CODE REVIEW

**Comment by noeliaSD** [comment]
@saledjenic LGTM! I've tested and it works fine. Just only a minor comment in code!

File: ui/imports/shared/views/SearchResults.qml [hunk 1]
```diff
@@ -44,9 +41,17 @@ Item {
         username = ""
         userAlias = ""
         pubKey = ""
+        isAddedContact = false
     }
 
-    width: parent.width
+    function isContactAdded() {
+        let d = Utils.getContactDetailsAsJson(root.pubKey).isContact
```
[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/imports/shared/views/SearchResults.qml]: @noeliaSD yes, thank you, that's some leftover I forgot to remove, done.

**Review by noeliaSD — APPROVED** [review]
Perfect!!

**Review by jrainville — APPROVED** [review]

**Comment by Khushboo-dev-cpp** [comment]
Hey @saledjenic ! This PR breaks the fix I provided in https://github.com/status-im/status-desktop/issues/4623. IN order to reproduce, copy wallet address and paste it in the address box when sending a transaction, you will start too see a wing contact being retrieved, after this I can see the issue again

**Comment by saledjenic** [comment]
@Khushboo-dev-cpp yes, that's why I explicitly added you as reviewer here, cause I saw your related changes, but wasn't able to reproduce the issue you fixed. Cause your fix brought this issue, and this fix seems brought back issue you fixed. :)  So I am going to check now what's the issue on that side and how we can fix it.

**Comment by saledjenic** [comment]
@Khushboo-dev-cpp I handled that issue in the PR #4745, check it out.

**Pull request merged**